### ISSUE TITLE
Cond component uses Javascript truthfulness semantics

### ISSUE THREAD

**Issue by jonaengs** [pynecone-io/pynecone#678]
**Describe the bug**
In Python, empty lists are falsy, meaning they evaluate to `False` when coerced into a boolean value.
In Javascript on the other hand, empty arrays are truthy ([source](https://developer.mozilla.org/en-US/docs/Glossary/Falsy)). Pynecone's `Cond` component uses Javascript semantics, not Python's when when evaluating the `Cond` conditional expression.

This behavior is not immediately clear, and it is not mentioned in the documentation of the `Cond` component.

**Examples:**
```
def index() -> pc.Component:
    # One would expect the two list expressions to be practically equivalent, 
    # but in fact they have the opposite result.
    return pc.list(
        pc.cond([], pc.text("empty arr is truthy"), pc.text("empty arr is falsy")),
        pc.text("empty arr is truthy") if [] else pc.text("empty arr is falsy"),
    )
```

```
class State(pc.State):
    arr = []
    arr_member = ""
        
    def unsafe_compute(self):
        # Should only be called when we know arr[0] exists
        self.arr_member = self.arr[0]

def index() -> pc.Component:
    return pc.list(
        # We expect this pc.cond call to behave like 'if arr:..', hiding the button if the list is empty
        pc.cond(State.arr, pc.button("Compute", on_click=State.unsafe_compute)),
    )
``` 
If the user expects `pc.cond(State.arr, ...)` to behave like Python's `if arr: ...`, they will expect the button to only appear when len(arr) >= 1.

**Expected behavior**
I would expect either the `Cond` component to follow Python's truthfulness semantics completely, or a mention in the Cond docs of the fact that the component's condition is evaluated in Javascript, and what that entails. 

As far as I can tell, empty lists are the only values for which Python and Javascript's truthfulness logic diverges (except NaN -- which can't be included in JSON -- and objects where the `__bool__` method can return `False`).

**Comment by Alek99** [comment]
Got it yeah we compile it down to js but we should either 

1) make the logic map the python logic 
or 
2)make this clear it doesn't in the docs.

**Comment by jonaengs** [comment]
Exactly. I think the first option would be preferable wrt ease of use and intuition/user expectations. One of the main selling points of Pynecone is that users don't have to know or learn Javascript to create good websites. So requiring users to memorize Javascript's truthfulness and equality semantics, would be a snag -- albeit a minor one.

Simply importing this function into all files and then 
```javascript
const py_truthy = (val) => {
    return Array.isArray(val) ? val.length > 0 : !!val
}
```
and then changing the return statement in `format_cond` to `return wrap(f"py_truthy({cond}) ? {true_value} : {false_value}", "{")` would be sufficient to fix this particular error.

However, there are several other places where Javascript and Python semantics diverge. For example, the equality of lists: 

Python:
```python
# Python performs recursive element-wise comparison (
[1] == [1]  # True
[[1]] == [[1]]  # True
```
Javascript:
```javascript
// Javascript simply looks at the address
[1] === [1] // false
[1] == [1] // false
```

Plugging all these holes could be a real challenge.

**Comment by picklelo** [comment]
Fixed by #750
